crypto/http/http_client.c

```diff
@@ -1168,12 +1188,16 @@ int OSSL_HTTP_proxy_connect(BIO *bio, const char *server, const char *port,
 
     for (;;) {
         /* will not actually wait if timeout == 0 */
-        rv = BIO_wait(fbio, max_time);
-        if (rv <= 0) {
-            BIO_printf(bio_err, "%s: HTTP CONNECT %s\n", prog,
-                       rv == 0 ? "timed out" : "failed waiting for data");
-            goto end;
+#ifndef OPENSSL_NO_SOCK
+        if (issocket) {
+            rv = BIO_wait(fbio, max_time);
```

Comment:
Also here we now have the drawback of a busy loop if the I/O is not socket-based.

Reply:
This issue should at least be mentioned in the documentation.